flax/linen/attention.py

```diff
@@ -258,31 +258,36 @@ def __call__(self,
                                    jnp.zeros, value.shape, value.dtype)
       cache_index = self.variable('cache', 'cache_index',
                                   lambda: jnp.array(0, dtype=jnp.int32))
+
       if is_initialized:
         *batch_dims, max_length, num_heads, depth_per_head = (
             cached_key.value.shape)
         # shape check of cached keys against query input
         expected_shape = tuple(batch_dims) + (1, num_heads, depth_per_head)
-        if expected_shape != query.shape:
+        cur_index = cache_index.value
+        # first input prompt can have sequence length > 1
+        if cur_index > 0 and cache_expected_shape != query.shape:
```

Comment:
`cur_index > 0` <-- we can't use value-dependent python control flow in JAX, as this value is an abstract tracer during trace+compilation.